PortalSwift/Sources/PortalSwift/Storage/Passkey/PasskeyStorage.swift

```diff
@@ -0,0 +1,266 @@
+  public var portalApi: PortalApi?
+  public var relyingParty: String
+  public var webAuthnHost: String
+
+  private var auth: PasskeyAuth
+  private let decoder = JSONDecoder()
+  private let logger = PortalLogger()
+  private var passkeyApi: HttpRequester
+  private let requests: PortalRequests
+  private var sessionId: String?
+
+  deinit {
+    print("PasskeyStorage is being deallocated")
+  }
+
+  public init(
+    relyingParty: String? = "portalhq.io",
+    webAuthnHost: String? = "backup.web.portalhq.io",
+    auth: PasskeyAuth? = nil,
+    encryption: PortalEncryption? = nil,
+    requests: PortalRequests? = nil
+  ) {
+    self.relyingParty = relyingParty ?? "portalhq.io"
+    self.auth = auth ?? PasskeyAuth(domain: self.relyingParty)
+    self.encryption = encryption ?? PortalEncryption()
+    self.requests = requests ?? PortalRequests()
+    self.webAuthnHost = "https://" + (webAuthnHost ?? "backup.web.portalhq.io")
+    self.passkeyApi = HttpRequester(baseUrl: self.webAuthnHost)
+  }
+
+  @available(*, deprecated, renamed: "PortalStorage", message: "Please use the new initialization pattern excluding your viewController.")
+  public init(
+    viewController: UIViewController? = nil,
+    relyingParty: String? = "portalhq.io",
+    webAuthnHost: String? = "backup.web.portalhq.io",
+    auth: PasskeyAuth? = nil,
+    encryption: PortalEncryption? = nil,
+    requests: PortalRequests? = nil
+  ) {
+    self.relyingParty = relyingParty ?? "portalhq.io"
+    self.auth = auth ?? PasskeyAuth(domain: self.relyingParty)
+    self.encryption = encryption ?? PortalEncryption()
+    self.requests = requests ?? PortalRequests()
+    self.webAuthnHost = "https://" + (webAuthnHost ?? "backup.web.portalhq.io")
+    self.passkeyApi = HttpRequester(baseUrl: self.webAuthnHost)
+
+    if let view = viewController {
+      self.auth.authenticationAnchor = view.view.window
+    }
+
+    super.init()
+  }
+
+  /*******************************************
+   * Public functions
+   *******************************************/
+
+  public func delete() async throws -> Bool {
+    throw StorageError.mustExtendStorageClass
+  }
+
+  public func read() async throws -> String {
+    guard let apiKey = self.apiKey else {
+      throw PasskeyStorageError.noApiKey
+    }
+
+    if let url = URL(string: "\(webAuthnHost)/passkeys/begin-login") {
+      let payload = ["relyingParty": relyingParty]
+      let data = try await requests.post(url, withBearerToken: apiKey, andPayload: payload)
+      let result = try decoder.decode(WebAuthnAuthenticationOption.self, from: data)
+
+      self.sessionId = result.sessionId
+
+      let assertion = try await withCheckedThrowingContinuation { [weak self] continuation in
+        self?.auth.continuation = continuation
+
+        DispatchQueue.main.async { [self] in
+          if let _ = self?.auth.authenticationAnchor {
+            self?.auth.signInWith(result.options, preferImmediatelyAvailableCredentials: true)
```

Comment:
do we need to add a check that the anchor is set before we call this function? Either here or in `signInWith` similar to how we did in gdrive with the config error check